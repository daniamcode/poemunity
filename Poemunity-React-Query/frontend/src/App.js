import React, { useState } from 'react'
import { Route, Switch, BrowserRouter as Router } from 'react-router-dom'
import './App.scss'
import Dashboard from './components/Dashboard'
import Detail from './components/Detail'
import Header from './components/Header'
import Profile from './components/Profile'
import Login from './components/Login'
import Register from './components/Register'
import PageNotFound from './components/PageNotFound'

export const AppContext = React.createContext();

function App (props) {
  const [contextState, setContextState] = useState({
    elementToEdit: '',
    user: null,
    userId: '',
    username: '',
    config: {},
    adminId: process.env.NODE_ENV === 'development' 
      ? process.env.REACT_APP_ADMIN_PRE
      : process.env.REACT_APP_ADMIN,
    setState(data) {
        const { setState, ...res } = data;
        const newState = { ...contextState, ...res };
        setContextState(newState);
    },
  });
  return (
    <AppContext.Provider value={contextState}>
      <Router>
        <div className='container'>
          <Header />

          <div className='margin-body'>
            <Switch>
              <Route path='/' exact component={Dashboard} />
              <Route path='/profile' component={Profile} />
              <Route path='/login' component={Login} />
              <Route path='/register' component={Register} />
              <Route path='/:genre' exact component={Dashboard} />
              <Route path='/detail/:poemId' exact component={Detail} />
              <Route component={PageNotFound} />
            </Switch>
          </div>
        </div>
      </Router>
    </AppContext.Provider>

  )
}


export default App
