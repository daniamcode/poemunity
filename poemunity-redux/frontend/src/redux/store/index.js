import { applyMiddleware, createStore } from 'redux';
import { createLogger }                 from 'redux-logger';
import thunk                            from 'redux-thunk';
import { composeWithDevTools }          from 'redux-devtools-extension';

import rootReducer      from '../reducers';

const { NODE_ENV } = process.env


// eslint-disable-next-line import/no-mutable-exports
let configureStore;
if (NODE_ENV !== 'production') {
    const enhancer = composeWithDevTools(
        applyMiddleware(thunk, createLogger()),
    );

    // eslint-disable-next-line no-shadow
    configureStore = function configureStore(initialState) {
        return createStore(rootReducer, initialState, enhancer);
    };
} else {
    // eslint-disable-next-line no-shadow
    configureStore = function configureStore(initialState) {
        return createStore(rootReducer, initialState, applyMiddleware(thunk));
    };
}

export default configureStore;
