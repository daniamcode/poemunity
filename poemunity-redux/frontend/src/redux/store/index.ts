import { configureStore } from '@reduxjs/toolkit'
import {rootReducer}      from '../reducers/rootReducer';
import { useDispatch } from 'react-redux'


const store = configureStore({
    reducer: rootReducer,
  })

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
export const useAppDispatch: () => AppDispatch = useDispatch // Export a hook that can be reused to resolve types
export default store


// import { applyMiddleware, createStore } from 'redux';
// import { createLogger }                 from 'redux-logger';
// import thunk                            from 'redux-thunk';
// import { composeWithDevTools }          from 'redux-devtools-extension';

// const { NODE_ENV } = process.env

// let configureStore
// if (NODE_ENV !== 'production') {
//     const enhancer = composeWithDevTools(
//         applyMiddleware(thunk, createLogger()),
//     );

//     configureStore = function configureStore(initialState) {
//         return createStore(rootReducer, initialState, enhancer);
//     };
// } else {
//     configureStore = function configureStore(initialState) {
//         return createStore(rootReducer, initialState, applyMiddleware(thunk));
//     };
// }

// export default configureStore;

