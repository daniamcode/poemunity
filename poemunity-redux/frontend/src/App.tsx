import React, { useState } from 'react'
import { Route, Switch, BrowserRouter as Router } from 'react-router-dom'
import './App.scss'
import Dashboard from './components/Dashboard'
import Detail from './components/Detail'
import Header from './components/Header'
import Profile from './components/Profile'
import Login from './components/Login'
import Register from './components/Register'
import PageNotFound from './components/PageNotFound'
import { Context } from './typescript/interfaces'

export const AppContext = React.createContext<Context>({
    elementToEdit: '',
    user: '',
    userId: '',
    username: '',
    picture: '',
    config: {},
    adminId: '',
    setState: () => {},
})

function App () {
  const [contextState, setContextState] = useState({
    elementToEdit: '',
    user: '',
    userId: '',
    username: '',
    picture: '',
    config: {},
    adminId: process.env.REACT_APP_ADMIN ?? '',
    setState(data: Context) {
      const { setState, ...res } = data;
      const newState = { ...contextState, ...res };
      setContextState(newState);
    },
  });

  return (
    <AppContext.Provider value={contextState}>
      <Router>
        <div className='container'>
          <Header />

          <div className='margin-body'>
            <Switch>
              <Route path='/' exact component={Dashboard} />
              <Route path='/profile' component={Profile} />
              <Route path='/login' component={Login} />
              <Route path='/register' component={Register} />
              <Route path='/:genre' exact component={Dashboard} />
              <Route path='/detail/:poemId' exact component={Detail} />
              <Route component={PageNotFound} />
            </Switch>
          </div>
        </div>
      </Router>
    </AppContext.Provider>

  )
}


export default App
