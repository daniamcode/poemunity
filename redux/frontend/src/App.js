import React from 'react'
import { Route, Switch, BrowserRouter as Router } from 'react-router-dom'
import './App.scss'
import Dashboard from './components/Dashboard'
import Detail from './components/Detail'
import Header from './components/Header'
import Profile from './components/Profile'
import PageNotFound from './components/PageNotFound'

function App (props) {
  return (
    <Router>
      <div className='container'>
        <Header />

        <div className='margin-body'>
          <Switch>
            <Route path='/' exact component={Dashboard} />
            <Route path='/perfil' component={Profile} />
            <Route path='/:genre' exact component={Dashboard} />
            <Route path='/detalle/:poemId' exact component={Detail} />
            <Route component={PageNotFound} />
          </Switch>
        </div>
      </div>
    </Router>
  )
}


export default App
