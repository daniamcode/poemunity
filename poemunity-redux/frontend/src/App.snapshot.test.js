import renderer from 'react-test-renderer'
import React from 'react'
import App from './App'
import {
  QueryClient,
  QueryClientProvider,
} from "react-query"
import { Provider } from 'react-redux';
import store from './redux/store';


describe('App snapshot', () => {
  const queryClient = new QueryClient();

  const tree = renderer.create(
    <Provider store={store}>
    <QueryClientProvider client={queryClient}>
      <App />
    </QueryClientProvider>
    </Provider>
  )

  test('should match', () => {
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
