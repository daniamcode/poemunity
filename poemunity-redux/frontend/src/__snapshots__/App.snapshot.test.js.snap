// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should match 1`] = `
<div
  className="container"
>
  <section
    className="header"
  >
    <div
      className="header__dropdown"
    >
      <div
        className="MuiPaper-root MuiAccordion-root accordion MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          aria-controls="panel1a-content"
          aria-disabled={false}
          aria-expanded={false}
          className="MuiButtonBase-root MuiAccordionSummary-root"
          id="panel1a-header"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiAccordionSummary-content"
          >
            <div
              className="header__dropdown-categories-icon"
            />
            <p
              className="header__dropdown-categories"
            >
              Categories
            </p>
          </div>
          <div
            aria-disabled={false}
            aria-hidden={true}
            className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role={null}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          className="MuiCollapse-root MuiCollapse-hidden"
          style={
            Object {
              "minHeight": "0px",
            }
          }
        >
          <div
            className="MuiCollapse-wrapper"
          >
            <div
              className="MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel1a-header"
                id="panel1a-content"
                role="region"
              >
                <div>
                  <div
                    className="MuiAccordionDetails-root"
                  >
                    <a
                      className="header__dropdown-subcategories"
                      href="/anniversary"
                      onClick={[Function]}
                    >
                      Anniversary
                    </a>
                  </div>
                  <div
                    className="MuiAccordionDetails-root"
                  >
                    <a
                      className="header__dropdown-subcategories"
                      href="/friendship"
                      onClick={[Function]}
                    >
                      Friendship
                    </a>
                  </div>
                  <div
                    className="MuiAccordionDetails-root"
                  >
                    <a
                      className="header__dropdown-subcategories"
                      href="/funny"
                      onClick={[Function]}
                    >
                      Funny
                    </a>
                  </div>
                  <div
                    className="MuiAccordionDetails-root"
                  >
                    <a
                      className="header__dropdown-subcategories"
                      href="/haikus"
                      onClick={[Function]}
                    >
                      Haikus
                    </a>
                  </div>
                  <div
                    className="MuiAccordionDetails-root"
                  >
                    <a
                      className="header__dropdown-subcategories"
                      href="/hope"
                      onClick={[Function]}
                    >
                      Hope
                    </a>
                  </div>
                  <div
                    className="MuiAccordionDetails-root"
                  >
                    <a
                      className="header__dropdown-subcategories"
                      href="/life"
                      onClick={[Function]}
                    >
                      Life
                    </a>
                  </div>
                  <div
                    className="MuiAccordionDetails-root"
                  >
                    <a
                      className="header__dropdown-subcategories"
                      href="/love"
                      onClick={[Function]}
                    >
                      Love
                    </a>
                  </div>
                  <div
                    className="MuiAccordionDetails-root"
                  >
                    <a
                      className="header__dropdown-subcategories"
                      href="/sad"
                      onClick={[Function]}
                    >
                      Sad
                    </a>
                  </div>
                </div>
                <div
                  className="MuiAccordionDetails-root"
                >
                  <a
                    className="header__dropdown-subcategories"
                    href="/"
                    onClick={[Function]}
                  >
                    All
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="header__logo"
    >
      <a
        className="header__text-logo-first"
        href="/"
        onClick={[Function]}
      >
        P
      </a>
      <a
        className="header__logo-icon"
        href="/"
        onClick={[Function]}
      />
      <a
        className="header__text-logo-second"
        href="/"
        onClick={[Function]}
      >
        emunity
      </a>
    </div>
    <p
      className="list__presentation"
    >
      Your poem community!!!
    </p>
    <div
      className="separator"
    />
    <button
      className="header__login"
      onClick={[Function]}
    />
  </section>
  <div
    className="margin-body"
  >
    <main
      className="dashboard"
    >
      <div
        className="dashboard__accordion"
      >
        <div
          className="MuiPaper-root MuiAccordion-root accordion MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            aria-controls="panel1a-content"
            aria-disabled={false}
            aria-expanded={false}
            className="MuiButtonBase-root MuiAccordionSummary-root"
            id="panel1a-header"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiAccordionSummary-content"
            >
              <div
                className="header__dropdown-categories-icon"
              />
              <p
                className="header__dropdown-categories"
              >
                Categories
              </p>
            </div>
            <div
              aria-disabled={false}
              aria-hidden={true}
              className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role={null}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            className="MuiCollapse-root MuiCollapse-hidden"
            style={
              Object {
                "minHeight": "0px",
              }
            }
          >
            <div
              className="MuiCollapse-wrapper"
            >
              <div
                className="MuiCollapse-wrapperInner"
              >
                <div
                  aria-labelledby="panel1a-header"
                  id="panel1a-content"
                  role="region"
                >
                  <div>
                    <div
                      className="MuiAccordionDetails-root"
                    >
                      <a
                        className="header__dropdown-subcategories"
                        href="/anniversary"
                        onClick={[Function]}
                      >
                        Anniversary
                      </a>
                    </div>
                    <div
                      className="MuiAccordionDetails-root"
                    >
                      <a
                        className="header__dropdown-subcategories"
                        href="/friendship"
                        onClick={[Function]}
                      >
                        Friendship
                      </a>
                    </div>
                    <div
                      className="MuiAccordionDetails-root"
                    >
                      <a
                        className="header__dropdown-subcategories"
                        href="/funny"
                        onClick={[Function]}
                      >
                        Funny
                      </a>
                    </div>
                    <div
                      className="MuiAccordionDetails-root"
                    >
                      <a
                        className="header__dropdown-subcategories"
                        href="/haikus"
                        onClick={[Function]}
                      >
                        Haikus
                      </a>
                    </div>
                    <div
                      className="MuiAccordionDetails-root"
                    >
                      <a
                        className="header__dropdown-subcategories"
                        href="/hope"
                        onClick={[Function]}
                      >
                        Hope
                      </a>
                    </div>
                    <div
                      className="MuiAccordionDetails-root"
                    >
                      <a
                        className="header__dropdown-subcategories"
                        href="/life"
                        onClick={[Function]}
                      >
                        Life
                      </a>
                    </div>
                    <div
                      className="MuiAccordionDetails-root"
                    >
                      <a
                        className="header__dropdown-subcategories"
                        href="/love"
                        onClick={[Function]}
                      >
                        Love
                      </a>
                    </div>
                    <div
                      className="MuiAccordionDetails-root"
                    >
                      <a
                        className="header__dropdown-subcategories"
                        href="/sad"
                        onClick={[Function]}
                      >
                        Sad
                      </a>
                    </div>
                  </div>
                  <div
                    className="MuiAccordionDetails-root"
                  >
                    <a
                      className="header__dropdown-subcategories"
                      href="/"
                      onClick={[Function]}
                    >
                      All
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="dashboard__list"
      >
        <div
          className="list__container"
        >
          <div
            className="list__intro"
          >
            <div
              className="list__search"
            >
              <div
                className="separator"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                style={
                  Object {
                    "fill": "#4F5D73",
                    "fontSize": 40,
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
              <div
                className="MuiFormControl-root MuiTextField-root"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                  data-shrink={false}
                  style={
                    Object {
                      "color": "#4F5D73",
                    }
                  }
                >
                  Search an author
                </label>
                <div
                  className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "#4F5D73",
                    }
                  }
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiInput-input"
                    disabled={false}
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                  />
                </div>
              </div>
            </div>
            <form
              className="list__sort"
            >
              <label>
                Authors:
                <select
                  id="origin"
                  name="origin"
                  onChange={[Function]}
                >
                  <option
                    selected={true}
                    value="all"
                  >
                    All
                  </option>
                  <option
                    selected={false}
                    value="famous"
                  >
                    Famous
                  </option>
                  <option
                    selected={false}
                    value="user"
                  >
                    Users
                  </option>
                </select>
              </label>
            </form>
            <form
              className="list__sort"
            >
              <label>
                Order poems by: 
                <select
                  id="sort"
                  name="sort"
                  onChange={[Function]}
                >
                  <option
                    selected={false}
                    value="Likes"
                  >
                    Likes
                  </option>
                  <option
                    selected={false}
                    value="Date"
                  >
                    Date
                  </option>
                  <option
                    selected={false}
                    value="Random"
                  >
                    Random
                  </option>
                  <option
                    selected={false}
                    value="Title"
                  >
                    Title
                  </option>
                </select>
              </label>
            </form>
          </div>
        </div>
      </div>
      <div
        className="dashboard__ranking"
      >
        <main
          className="ranking"
        >
          <h3
            className="ranking__title"
          >
            Poets ranking
          </h3>
          <h5
            className="ranking__subtitle"
          >
            3 points per poem, 1 per like
          </h5>
          <div
            className="MuiPaper-root MuiTableContainer-root ranking__body MuiPaper-elevation1 MuiPaper-rounded"
          >
            <table
              aria-label="simple table"
              className="MuiTable-root makeStyles-table-1"
              role={null}
            >
              <thead
                className="MuiTableHead-root"
                role={null}
              >
                <tr
                  className="MuiTableRow-root MuiTableRow-head"
                  role={null}
                >
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                    scope="col"
                  >
                    Poets
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                    scope="col"
                  >
                    Points
                  </th>
                </tr>
              </thead>
              <tbody
                className="MuiTableBody-root"
                role={null}
              />
            </table>
          </div>
        </main>
      </div>
    </main>
  </div>
</div>
`;
